engine/server/session.go

```diff
@@ -322,7 +327,21 @@ func (srv *Server) initializeDaggerClient(
 	failureCleanups *buildkit.Cleanups,
 	opts *ClientInitOpts,
 ) error {
-	// initialize all the buildkit state for the client
+	// TODO: this is wrong right? Check with Alex
```

Comment:
(apologies for the numerous pings but) cc @vito, I noticed that we were still using this spanCtx in some places but I can't see that there's anything in the context to give a span. Maybe there is some setup somewhere that I'm missing?

TBH I haven't really done my homework here, I just left this TODO since there was too much other crap to do, so if you aren't immediately sure just let me know and I'll dig around more.

Reply:
All good on the pings, sorry I haven't given this all due diligence yet!

From a quick skim on `main` it seems like `spanCtx` is never actually initialized. I always see `traceID=000000000000000000` in the Buildkit logs too. So, any change you make here is probably better than the current state.

I think this was purely in service of getting trace propagation to work for socket and registry auth session attachables, which might change fundamentally (for the better) with this PR, so it seems like the appropriate time to fix them up.

The best way to test that is to send telemetry to Honeycomb and see if the trace ends up with spans that have missing parents. (It'll put a warning up top.)